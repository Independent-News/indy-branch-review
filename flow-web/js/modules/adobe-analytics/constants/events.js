export const INTERNAL_SEARCH = 'event1';
export const TOPIC_PAGE_LOAD = 'event2';
export const GALLERY_VIEW = 'event3';
export const AUTO_REFRESH = 'event6';
export const ARTICLE_VIEW = 'event7';
export const SELLING_PAGE_LOADED = 'event8';
export const SIGN_IN = 'event9';
export const ARTICLE_SHARE = 'event10';
export const COMMENT_ADD = 'event12';
export const READ_MORE_CLICK = 'event13';
export const AUTO_GALLERY_VIEW_COUNT = 'event14';
export const PREMIUM_COMMENT_ADD = 'event16';
export const ARTICLE_BOOKMARKED = 'event17';
export const RHR_PREMIUM_COMPONENT_CLICK = 'event18';
export const RHR_MOST_POPULAR_CLICK = 'event19';
export const RHR_INDY100_TRENDING_CLICK = 'event20';
export const SUBSCRIBE_BUTTON_CLICK = 'event21';
export const REG_PAGE_LOAD = 'event22';
export const REG_SUCCESS = 'event23';
export const MARKETING_OPT_IN_CLICK = 'event24';
export const PAYMENT_DETAILS_LOADED = 'event25';
export const PROMO_CODE_APPLIED = 'event26';
export const PAYMENT_DETAILS_SUBMITTED = 'event27';
export const SUBSCRIPTION_SUCCESS = 'event28';
export const AUTO_RENEW_DISABLED = 'event29';
export const RHR_POPULAR_VIDEOS_CLICK = 'event30';
export const RHR_SPONSORED_FEATURES_CLICK = 'event31';
export const PAYMENT_UNSUCCESSFUL = 'event32';
export const DONATION_BUTTON_CLICK = 'event34';
export const TAX_CONFIRMATION_PROMPT = 'event35';
export const DONATION_ARTICLE_BOTTOM_CLICK = 'event36';
export const NAV_CLICK = 'event37';
export const ARTICLE_INLINE_LINK_CLICK = 'event39';
export const ARTICLE_TOPIC_TAG_CLICK = 'event40';
export const ARTICLE_SHARE_ICON_CLICK = 'event41';
export const COMMENTS_SHOW_CLICK = 'event42';
export const COMMENTS_TAB_CLICK = 'event43';
export const NOTIFICATION_PROMPT_LOAD = 'event44';
export const NOTIFICATION_PROMPT_ACCEPT = 'event45';
export const NOTIFICATION_PROMPT_DISMISS = 'event46';
export const HOMEPAGE_LINK_CLICK = 'event46'; // this is used, but Graham isn't sure what for
export const MOST_WATCHED_CLICK = 'event51';
export const COMMENTS_LOAD_MORE = 'event53';
export const RELATED_ARTICLE_BULLET_CLICK = 'event57';
export const LIVE_BLOG_INLINE_ARTICLE_CLICK = 'event58';
export const LIVE_BLOG_KEY_POINTS_CLICK = 'event60';
export const REG_GATE_SHOWN = 'event62';
export const NEWSLETTER_LITE_REG_SIGNUP = 'event63';
export const NEWSLETTER_PREFS_SUBSCRIBE = 'event64';
export const NEWSLETTER_PREFS_UNSUBSCRIBE = 'event65';
export const NEWSLETTER_PAGE_SIGNUP_CLICK = 'event66';
export const REGISTER_NOW_CLICK = 'event68';
export const EBOOKS_CLICK = 'event69';
export const PREMIUM_EVENT_CLICK = 'event70';
export const ARTICLE_INLINE_LINK_TOPIC_PAGE = 'event72';
export const ARTICLE_INLINE_LINK_NON_TOPIC_PAGE = 'event73';
export const SUBSCRIPTION_FORM_FIELD_FOCUS = 'event74';
export const SUBSCRIPTION_METHOD_SELECTED = 'event75';
export const SUBSCRIPTION_PACKAGE_CHANGED = 'event76';
export const MOST_COMMENTED_SECTION_CLICK = 'event78';
export const AFFILIATE_LINK_CLICK = 'event79';
export const BURGER_MENU_CLICK = 'event80';
export const EDITION_LINK_CLICK = 'event81';
export const VOUCHER_TICKER_LINK_CLICK = 'event82';
export const VOUCHER_TICKER_ARROW_CLICK = 'event83';
export const PIANO_EXPERIENCE_IMPRESSION = 'event84';
export const ACCOUNT_OVERVIEW_CLICK = 'event85';
export const CANCEL_JOURNEY_CLICK = 'event86';
export const CANCELLATION_JOURNEY_STAGE = 'event87';
export const WEBPUSH_SIGNUP_CLICK = 'event106';

export const BUY_NOW_CLICK = 'event93';
export const PRICE_COMPARISON_BUY_NOW_CLICK = 'event107';
export const PRODUCT_CAROUSEL_LOADED = 'event97';
export const PRODUCT_CAROUSEL_BUY_NOW_CLICK = 'event108';
export const PRODUCT_CAROUSEL_READ_REVIEW_CLICK = 'event99';
export const PRODUCT_COMPARISON_CHECKBOX_SELECTED = 'event100';
export const PRODUCT_COMPARISON_MODAL_OPEN = 'event101';

export const ADBLOCK_DETECTED = 'event105';

export const SUPPORT_NSC_LINK_CLICK = 'event110';
export const DONATE_PAGE_LOAD = 'event110';
