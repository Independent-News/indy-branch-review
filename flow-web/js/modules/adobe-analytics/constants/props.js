export const SUB_SECTION = 'prop1';
export const SUB_SECTION_2 = 'prop2';
export const DOMAIN = 'prop3';
export const INTERNAL_SEARCH_TERMS = 'prop4';
export const INDEPENDENT_PREMIUM_ACCOUNT_PORTAL = 'prop5';
export const NAV_LINK_CLICK = 'prop6';
export const NAV_LINK_CLICK_LOCATION = 'prop7';
export const PAGE_TYPE = 'prop8';
export const ARTICLE_ID = 'prop9';
export const ARTICLE_PUBLICATION_TIME = 'prop10';
export const HEADLINES = 'prop11';
export const ARTICLE_NAME_OR_ID = 'prop12';
export const BYLINE = 'prop13';
export const TOPIC_PAGES = 'prop14';
export const LEAD_MEDIA_ITEM = 'prop15';
export const ORIG_PUB_DATE = 'prop16';
export const BOOKMARKED_ARTICLE_ID = 'prop17';
export const ARTICLE_TYPE = 'prop19';
export const CONTENT_SOURCE = 'prop20';
export const GALLERY_VIEW_IMAGE_NO = 'prop21';
export const WORD_COUNT = 'prop22';
export const INTERNAL_LINK_COUNT = 'prop23';
export const COMPONENTS_LIST = 'prop24';
export const PRODUCT_COMPARISON_FAILURE = 'prop25';
export const RELATED_ARTICLES_VIDEO_ALGORITHM = 'prop26';
export const RELATED_LINK_POSITION = 'prop27';
export const ARTICLE_MOD_DATE = 'prop28';
export const AUTO_LEAD_MEDIA_SWAP = 'prop30';
export const EDITION_LOCALE = 'prop31';
export const BURGER_MENU_CALL_TO_ACTION = 'prop32';
export const TOP_NAV_LINK_TYPE = 'prop34';
export const PLATFORM = 'prop36';
export const NEWSLETTER = 'prop37';
export const ARTICLE_TOPIC_TAGS = 'prop38';
export const COMMERCIAL_CAMPAIGN = 'prop39';
export const URL = 'prop40';
export const LINK_DESTINATION = 'prop41';
export const LINK_LABEL = 'prop42';
export const PACKAGE = 'prop43';
export const SUBSCRIPTION_LENGTH = 'prop44';
export const SUBSCRIPTION_PRICE = 'prop45';
export const PARAGRAPH = 'prop46';
export const VALID_CODE = 'prop47';
export const OPT_IN_OUT = 'prop48';
export const HUB_VIDEO_TYPE = 'prop49';
export const SUBSCRIBE_BUTTON_VERSION = 'prop50';
export const CODE_VERSION = 'prop64';
export const REPUBLICATION_DATE = 'prop65';
export const ARTICLE_PREMIUM_STATUS = 'prop66';
export const LINK_POSITION = 'prop67';
export const IPHONE_SCREEN_SIZE = 'prop68';
export const USER_LOGGED_IN_STATUS = 'prop69';
export const USER_ID = 'prop70';
export const USER_REG_DATE = 'prop71';
export const USER_SHARE_PROVIDER = 'prop72';
export const USER_STATUS = 'prop75';
export const INITIAL_SELECTED_PACKAGE = 'prop90';
