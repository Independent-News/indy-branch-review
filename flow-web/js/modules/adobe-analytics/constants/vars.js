export const SUB_SECTION = 'eVar1';
export const SUB_SECTION_2 = 'eVar2';
export const DOMAIN = 'eVar3';
export const INTERNAL_SEARCH_TERMS = 'eVar4';
export const INDY_PREMIUM_ACCOUNT_PORTAL = 'eVar5';
export const NAV_LINK_CLICK = 'eVar6';
export const NAV_LINK_CLICK_LOCATION = 'eVar7';
export const PAGE_TYPE = 'eVar8';
export const ARTICLE_ID = 'eVar9';
export const ARTICLE_PUBLICATION_TIME = 'eVar10';
export const HEADLINES = 'eVar11';
export const SITE_SECTIONS = 'eVar12';
export const BYLINE = 'eVar13';
export const TOPIC_PAGES = 'eVar14';
export const LEAD_MEDIA_ITEM = 'eVar15';
export const PAGE_PATH = 'eVar16';
export const PAGE_QUERY = 'eVar17';
export const ARTICLE_TYPE = 'eVar19';
export const CONTENT_SOURCE = 'eVar20';
export const PREV_PAGE_URL = 'eVar21';
export const WORD_COUNT = 'eVar22';
export const INTERNAL_LINK_COUNT = 'eVar23';
export const COMPONENTS_LIST = 'eVar24';
export const YAHOO_ARTICLE_ID = 'eVar25';
export const RELATED_ARTICLES_VIDEO_ALGORITHM = 'eVar26';
export const RELATED_LINK_POSITION = 'eVar27';
export const ARTICLE_MOD_DATE = 'eVar28';
export const EDITION_LOCALE = 'eVar31';
export const PLATFORM = 'eVar36';
export const NEWSLETTER = 'eVar37';
export const ARTICLE_TOPIC_TAGS = 'eVar38';
export const COMMERCIAL_CAMPAIGN = 'eVar39';
export const URL = 'eVar40';
export const LINK_DESTINATION = 'eVar41';
export const LINK_LABEL = 'eVar42';
export const PACKAGE = 'eVar43';
export const SUBSCRIPTION_LENGTH = 'eVar44';
export const SUBSCRIPTION_PRICE = 'eVar45';
export const PARAGRAPH = 'eVar46';
export const VALID_CODE = 'eVar47';
export const OPT_IN_OUT = 'eVar48';
export const HUB_VIDEO_TYPE = 'eVar49';
export const SUBSCRIBE_BUTTON_VERSION = 'eVar50';
export const UTM_SOURCE = 'eVar51';
export const UTM_MEDIUM = 'eVar52';
export const UTM_CAMPAIGN = 'eVar53';
export const UTM_TERM = 'eVar54';
export const UTM_CONTENT = 'eVar55';
export const UTM_FULL_STRING = 'eVar56';
export const INTERNAL_CHANNEL = 'eVar57';
export const INTERNAL_CAMPAIGN_NAME = 'eVar58';
export const INTERNAL_CAMPAIGN_DATE = 'eVar59';
export const INTERNAL_CAMPAIGN_AUDIENCE = 'eVar60';
export const INTERNAL_CAMPAIGN_CONTENT = 'eVar61';
export const IID_FULL_STRING = 'eVar62';
export const INTERNAL_CHANNEL_QS_PARAM = 'eVar63';
export const CODE_VERSION = 'eVar64';
export const LINK_POSITION = 'eVar67';
export const IPHONE_SCREEN_SIZE = 'eVar68';
export const USER_LOGGED_IN_STATUS = 'eVar69';
export const USER_ID = 'eVar70';
export const USER_REG_DATE = 'eVar71';
export const USER_SHARE_PROVIDER = 'eVar72';
export const USER_STATUS = 'eVar75';
export const UTM_MEDIUM_FIRST_TOUCH = 'eVar76';
export const UTM_MEDIUM_LINEAR_TOUCH = 'eVar77';
export const UTM_SOURCE_FIRST_TOUCH = 'eVar78';
export const UTM_SOURCE_LINEAR_TOUCH = 'eVar79';
export const UTM_CAMPAIGN_FIRST_TOUCH = 'eVar80';
export const UTM_CAMPAIGN_LINEAR_TOUCH = 'eVar81';
export const INTERNAL_CHANNEL_FIRST_TOUCH = 'eVar82';
export const INTERNAL_CHANNEL_LINEAR_TOUCH = 'eVar83';
export const INTERNAL_CAMPAIGN_NAME_FIRST_TOUCH = 'eVar84';
export const INTERNAL_CAMPAIGN_NAME_LINEAR_TOUCH = 'eVar85';
export const IP_ADDRESS = 'eVar86';
export const QUERY_STRING = 'eVar87';
export const REG_GATE_TRIGGER = 'eVar89';
export const INITIAL_SELECTED_PACKAGE = 'eVar90';
export const FORM_FIELD_DETAILS = 'eVar91';
export const AFFILIATE_LINK_COUNT = 'eVar92';
export const EXPERIENCE_NAME = 'eVar93';
export const EXPERIENCE_VARIANT = 'eVar94';
export const SELLING_PAGE_VARIANT_CLICK = 'eVar100';
export const PRODUCT_NAME = 'eVar96';
export const PRODUCT_PRICE = 'eVar97';
