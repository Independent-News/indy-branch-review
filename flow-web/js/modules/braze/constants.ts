export const BRAZE_EVENT_AUTORENEWAL_ON = 'turned_auto_renewalON';
export const BRAZE_EVENT_AUTORENEWAL_OFF = 'turned_auto_renewalOFF';
export const BRAZE_EVENT_PREMIUM_METER_EXPIRED = 'premium_meter_expired';
export const BRAZE_EVENT_REGISTRATION_COMPLETE = 'registration_complete';
export const BRAZE_EVENT_SUBSCRIPTION_START = 'subscription_start';
export const BRAZE_EVENT_SUBSCRIPTION_FINISH = 'subscription_finish';
export const BRAZE_EVENT_SUBSCRIPTION_ONBOARDING_SURVEY =
  'subscription_onboarding_survey';
export const BRAZE_EVENT_DONATION_START = 'donation_start';
export const BRAZE_EVENT_DONATION_FINISH = 'donation_finish';
