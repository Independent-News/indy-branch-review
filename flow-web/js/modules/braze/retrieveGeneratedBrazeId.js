import { error } from '../log';

import retrieveBrazeDeviceId from './retrieveBrazeDeviceId';

export default async () => {
  try {
    const brazeDeviceId = await retrieveBrazeDeviceId();
    if (!brazeDeviceId) {
      throw new Error('Error: failed to retrieve Braze device id');
    }
    const res = await fetch(
      `/api/user/braze-generated-id?device_id=${brazeDeviceId}`,
    );
    const resBody = await res.json();
    if (res.ok) {
      return resBody.generatedBrazeId;
    }
    throw new Error(resBody.message);
  } catch (e) {
    error(`Unable to retrieve generated Braze id: ${e.message}`);
    return null;
  }
};
