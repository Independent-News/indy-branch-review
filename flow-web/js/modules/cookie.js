/** @deprecated
 *
 * use getCookie() instead from app/public/js/utils/cookies.js
 */
export const getCookie = (name, defaultValue = null) => {
  const value = `; ${document.cookie}`;
  const parts = value.split(`; ${name}=`);
  return parts.length === 2
    ? decodeURIComponent(parts.pop().split(';').shift())
    : defaultValue;
};

/**
 * @param {string} name
 * @param {string} value
 * @param {object} [expire=false]
 * @param {number} [expire.days=0]
 * @param {number} [expire.hours=1]
 */
export const setCookie = (name, value, expire) => {
  const date = new Date();

  if (expire) {
    const expireTime = expire.days
      ? expire.days * 24 * 60 * 60 * 1000
      : expire.hours * 60 * 60 * 1000;
    date.setTime(date.getTime() + expireTime);
  }

  const parts = [];
  parts.push(`${name}=${encodeURIComponent(value)}`);
  parts.push('path=/');
  parts.push('secure');
  expire && parts.push(`expires=${date.toGMTString()}`);
  document.cookie = parts.join(';');
};

export const removeCookie = (name) => {
  const assignment = `${name}=null`;
  const expires = `expires=${new Date().toGMTString()}`;
  const path = 'path=/';
  document.cookie = [assignment, expires, path].join(';');
};

export const hasCookie = function (name) {
  return getCookie(name) !== null;
};
