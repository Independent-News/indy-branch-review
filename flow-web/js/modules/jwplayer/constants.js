export const PARAM_AD_POS = 'ap';
