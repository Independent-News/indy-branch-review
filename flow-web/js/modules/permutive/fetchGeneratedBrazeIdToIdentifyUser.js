import retrieveGeneratedBrazeId from '../braze/retrieveGeneratedBrazeId';

import { identifyUserViaGeneratedBrazeIdOnceLoaded } from './identifyUser';

export default async (isLoggedIn = false, permutiveIds) => {
  if (!isLoggedIn) {
    return null;
  }
  // if Braze has already loaded
  if (window.braze) {
    return await retrieveGeneratedBrazeId();
  }
  identifyUserViaGeneratedBrazeIdOnceLoaded(permutiveIds);
  return null;
};
