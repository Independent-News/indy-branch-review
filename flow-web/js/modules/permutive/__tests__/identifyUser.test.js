/**
 * @jest-environment jsdom
 */

import {
  COOKIE_CHARTBEAT_ID,
  COOKIE_HASHED_EMAIL,
  COOKIE_PUID,
} from '#app/constants/cookies';

import { EVENT_BRAZE_LOADED } from '#app/constants/customEvents';

import retrieveGeneratedBrazeId from '../../braze/retrieveGeneratedBrazeId';
import { getCookie } from '../../cookie';
import fetchGeneratedBrazeIdToIdentifyUser from '../fetchGeneratedBrazeIdToIdentifyUser';
import {
  identifyUser,
  identifyUserViaGeneratedBrazeIdOnceLoaded,
} from '../identifyUser';

jest.mock('../../cookie');
jest.mock('../../braze/retrieveGeneratedBrazeId');
jest.mock('../fetchGeneratedBrazeIdToIdentifyUser');

describe('identify user - Permutive', () => {
  beforeEach(() => {
    window.permutive = {
      identify: jest.fn(),
    };
  });

  describe('identifyingUsers', () => {
    afterEach(() => {
      delete window.permutive;
      delete window.braze;
      getCookie.mockReset();
      jest.resetAllMocks();
      jest.restoreAllMocks();
    });

    it('should identify user via Piano id, Braze id, Chartbeat id and hashed email', async () => {
      const expectedLoggedIn = true;
      const expectedPermutiveIds = [
        {
          id: 'the-piano-id',
          priority: 0,
          tag: 'piano_id',
        },
        {
          id: 'boris@example.test',
          priority: 1,
          tag: 'user_email',
        },
        {
          id: 'boris@example.test',
          priority: 2,
          tag: 'email_sha256',
        },
        {
          id: 'the-chartbeat-id',
          priority: 4,
          tag: 'chartbeat_id',
        },
        {
          id: 'some-braze-id',
          priority: 3,
          tag: 'braze_id',
        },
      ];
      window.braze = {};
      getCookie.mockImplementation((name) => {
        return {
          [COOKIE_PUID]: 'the-piano-id',
          [COOKIE_HASHED_EMAIL]: 'boris@example.test',
          [COOKIE_CHARTBEAT_ID]: 'the-chartbeat-id',
        }[name];
      });
      fetchGeneratedBrazeIdToIdentifyUser.mockResolvedValueOnce(
        'some-braze-id',
      );
      await identifyUser();
      expect(window.permutive.identify.mock.calls).toMatchSnapshot();
      expect(fetchGeneratedBrazeIdToIdentifyUser).toHaveBeenCalledTimes(1);
      expect(fetchGeneratedBrazeIdToIdentifyUser).toHaveBeenCalledWith(
        expectedLoggedIn,
        expectedPermutiveIds,
      );
    });

    it('should set permutive generated id for guest', async () => {
      const expectedPermutiveIds = [
        {
          id: 'test-user-id',
          priority: 5,
          tag: 'permutive_id',
        },
      ];
      window.permutive.context = {
        user_id: 'test-user-id',
      };
      getCookie.mockReturnValue(null);
      await identifyUser();
      expect(window.permutive.identify).toHaveBeenCalledWith(
        expectedPermutiveIds,
      );
    });
  });

  describe('identifyUserViaGeneratedBrazeIdOnceLoaded()', () => {
    const mockPermutiveId = ['some-permutive-id'];

    afterEach(() => {
      jest.clearAllMocks();
    });

    it('will setup a listener for braze loaded event for logged in user', () => {
      const addEventListener = jest.spyOn(document, 'addEventListener');
      identifyUserViaGeneratedBrazeIdOnceLoaded(mockPermutiveId);
      expect(addEventListener).toHaveBeenCalledTimes(1);
      expect(addEventListener).toHaveBeenCalledWith(
        EVENT_BRAZE_LOADED,
        expect.any(Function),
      );
      expect(retrieveGeneratedBrazeId).not.toHaveBeenCalled();
    });

    it('will not track user via Permutive if generated Braze id is not found', async () => {
      retrieveGeneratedBrazeId.mockResolvedValueOnce(null);
      await identifyUserViaGeneratedBrazeIdOnceLoaded(mockPermutiveId);
      document.dispatchEvent(new CustomEvent(EVENT_BRAZE_LOADED));
      // puts the new promise at the end on microtask queue and holds the test execution until it is resolved. This allows for the async callback, which was put on the microtask queue before our promise, to be executed and make changes that we expect.
      await new Promise(process.nextTick);
      expect(retrieveGeneratedBrazeId).toHaveBeenCalled();
      expect(window.permutive.identify).not.toHaveBeenCalled();
    });

    it('track user via Permutive using generated Braze id on Braze load event', async () => {
      retrieveGeneratedBrazeId.mockResolvedValueOnce('some-braze-id');
      await identifyUserViaGeneratedBrazeIdOnceLoaded(mockPermutiveId);
      document.dispatchEvent(new CustomEvent(EVENT_BRAZE_LOADED));
      await new Promise(process.nextTick);
      expect(retrieveGeneratedBrazeId).toHaveBeenCalled();
      expect(window.permutive.identify).toHaveBeenCalledTimes(1);
      expect(window.permutive.identify).toHaveBeenCalledWith(
        expect.arrayContaining([
          {
            id: 'some-braze-id',
            priority: 3,
            tag: 'braze_id',
          },
        ]),
      );
    });
  });
});
