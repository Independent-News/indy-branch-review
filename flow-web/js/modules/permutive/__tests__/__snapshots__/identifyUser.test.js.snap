// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`identify user - Permutive identifyingUsers should identify user via Piano id, Braze id, Chartbeat id and hashed email 1`] = `
[
  [
    [
      {
        "id": "the-piano-id",
        "priority": 0,
        "tag": "piano_id",
      },
      {
        "id": "boris@example.test",
        "priority": 1,
        "tag": "user_email",
      },
      {
        "id": "boris@example.test",
        "priority": 2,
        "tag": "email_sha256",
      },
      {
        "id": "the-chartbeat-id",
        "priority": 4,
        "tag": "chartbeat_id",
      },
      {
        "id": "some-braze-id",
        "priority": 3,
        "tag": "braze_id",
      },
    ],
  ],
]
`;
